components/gfx/display_list/mod.rs

```diff
@@ -1043,6 +1043,22 @@ impl ClippingRegion {
     }
 }
 
+impl fmt::Debug for ClippingRegion {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        if self.main == max_rect() && self.complex.is_empty() {
```

Comment:
Given there are only two condition in all the if else chain, could we write this like:

```rust

match (self.main == max_rect(), self.complex.is_empty()) {
    (true, true) => write!(f, "ClippingRegion::Max"),
    // ...
}
```